llvm/lib/Target/SPIRV/SPIRVGlobalRegistry.cpp

```diff
@@ -424,7 +424,7 @@ Register SPIRVGlobalRegistry::getOrCreateCompositeOrNull(
     LLT LLTy = LLT::scalar(64);
     Register SpvVecConst =
         CurMF->getRegInfo().createGenericVirtualRegister(LLTy);
-    CurMF->getRegInfo().setRegClass(SpvVecConst, &SPIRV::iIDRegClass);
+    CurMF->getRegInfo().setRegClass(SpvVecConst, getRegClass(SpvType));
```

Comment:
NIT: Since this was a bug fix. Would it have been better to do this as a seperate commit to capture why this was necessary?

Reply:
Happy to put it in its own PR if that is what people want. I just left it here since it is such a small change.

Reply:
I imagine there are other places in the SPIRV backend that might require a fix like this and want to give a future person a hint at how to fix it.

Alternatively create a second issue which refrences the problem you found. Add it to the pr description as fixed by this pr. drop a link to the issue in this comment section. That way if anyone searches github issues for something similar to this they will find your issue and see how you fixed it.

Reply:
I created issue #113486 and mentioned it in this PR.